frontend/src/views/SignUp.vue

```diff
@@ -64,21 +101,43 @@
     </v-container>
   </v-sheet>
 </template>
-
 <script>
 export default {
   name: 'SignUp',
-
   data() {
     return {
       email: null,
       password: null,
       user: null,
       progress: false,
+      name: null,
+      bio: null,
+      website: null,
+      rules: [
+        value => !!value || 'Required.',
+        value => value.trim().length > 0 || 'Enter Valid Name or Username'],
+      passlength: [
+        value => value.trim().length > 0 || 'Required.',
+        value => (value || '').length >= 8 || 'min 8 characters'],
     };
   },
-
   methods: {
+    validate() {
+      if (this.$refs.signup().validate()) {
+        this.snackbar = true;
+      }
+    },
+    check() {
+      this.$firebase.database().ref().child('users').orderByChild('username')

```

Comment:
We're not using the database we're using Firestore. Look at every other example in the code for an example!